### ISSUE TITLE
Search, sort, sync setting and more feature request

### ISSUE THREAD

**Issue by isumairu** [Authenticator-Extension/Authenticator#140]
If you could fulfill these requests from me, that would be great.

1. Add search feature (I find Authenticator often fail to show only accounts of the site opened) and sort setting (I have many accounts and this extension seems to show them in no order. help me).
2. Add tooltips and more detailed description. What's dropbox's icon doing on the top bar? At what interval does Authenticator backup to dropbox? And other places that doesn't have explanation or hint. More infos wouldn't hurt don't you think?
3. In the export/Import setting, I suggest changing "Download" to "Save". It'd make more sense.
4. Add option to not sync to google (in chrome) and/or firefox.
5. Add option to manually start backup to dropbox.
6. Add option to launch Authenticator as a stand alone window (so it look like Authy desktop app).
7. I don't know if this is possible, but, can you make Authenticator import from dropbox or other cloud? automatically? I want synchronization between my chrome & firefox.

That's it for now. 
Thank you.

**Comment by mymindstorm** [comment]
1. See #138, In regards to sorting you can go into edit mode and drag them to change their order.
2. That's a good idea, but it will take a while to translate the strings for any tooltips. The Dropbox icon appears when Dropbox sync is enabled. It attempts to sync to Dropbox every 30 days, if you need more info here is the relevant source:
https://github.com/Authenticator-Extension/Authenticator/blob/87d812deb814baee9b83a9ddefe3d140df431a6e/src/popup.ts#L54-L65
https://github.com/Authenticator-Extension/Authenticator/blob/01c972ba6a3d0f17b6ead6239833eccca07824ef/src/models/dropbox.ts#L11-L64

3. Save is a bit ambiguous, whereas if we use download you know instantly that it will save a file to your downloads folder.
4. We rely on the sync storage space, if you don't want to sync it you need to turn off extension data sync in chrome sync settings if possible.
5. That is a good feature request
6. Being a WebExtnesion, that's not really in scope of this project. If you want to make an electron fork, feel free to do so.
7. Before we work on anything like that there are some other sync issues that need to be fixed.

**Comment by isumairu** [comment]
Thanks for your response.

Regarding your No. 1 reply, I'm aware of the drag-sort feature, but it's not enough nor practical for me. It can't even scroll continuously (I said I have many accounts, right?). Also, I still need search feature. Please.

Regarding No. 6, I have several question. 
1. Can you please explain to me why is that Authenticator being a WebExtension inhibit you from making it able to have standalone window? (separate tab will do as well) It just feels inconvenient for me to click the toolbar icon and then scroll to find my account every time I need it (I often need to login to many website quickly and in several browsers)
2. Or, can you make it become a chrome app?
3. Is Authenticator's WebExtension in Chrome differ much from WebExtension in Firefox?

Thank you.

**Comment by mymindstorm** [comment]
Our drag and drop support is very basic, the library we use does not support scrolling as you drag. Not much can be done about that, sorry.

I'm not against search, I was just thinking it feels a bit redundant with the auto-filter. Can you elaborate on where the auto-filter is failing you? Since you use the extension heavily you should be familiar with faults in filtering that most users don't notice or care about. Even if you still want search, I want to improve user experience for others. Dropping f-bombs and calling it useless will not help anyone here.

- WebExtensions aren't supposed to be standalone programs / windows, just extensions of the browser's functionality. If we did do something along those lines it would be better for us to use something like the [sidebar](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/user_interface/Sidebars). (not supported in chrome) I am strongly opposed to breaking UX, so I don't think we would ever add a button to do that. I will give you a workaround though.
- Chrome apps are not supported and cannot be created anymore. 
- I made the Firefox version, the original author and I now work together on it. The code is now the same, but everything before version 5.0.2 was on my fork. 

To open the extension in a tab in chrome open this link: `chrome-extension://bhghoamapcdpbohphigoooaddinpkbai/view/popup.html` In Firefox you need to go to `about:debugging` and get the "Internal UUID" of Authenticator. Then go to `moz-extension://[your UUID]/view/popup.html` You might want to pin or bookmark the tab.

**Comment by isumairu** [comment]
Who's dropping f-bombs? Who's calling something useless?

Elaborate on where the auto-filter is failing me? Your tips on adding ::host is helpful, but w/o decent sorting option, even with Authenticator showing only accounts from the right site, I still have to scroll and find my account among almost a hundred entries. Can you see how search feature will be helpful in this condition?

I didn't want Authenticator to be a standalone program, separated from the browser. I just want it to have a small window that doesn't disappear when I change focus. Several WebExtensions can do this, e.g. Video DownloadHelper. Your workaround is helpful though, thanks. 
Is it possible to open the popup.html in a pop-up window (minimum window with only an address bar).  Do you know how?

**Comment by mymindstorm** [comment]
You?
![image](https://user-images.githubusercontent.com/27789806/40749633-463f591c-642a-11e8-9a57-cbe7a91d193d.png)

> almost a hundred entries

By many I thought you had around 30 not even close to that many. Thanks for the suggestions. A pop-out mode and search would be useful.

---

Follow the [debugging steps](https://github.com/Authenticator-Extension/Authenticator/wiki/Getting-Debugging-Information) to open a console and paste this in for now. It should be exactly what you want.
```javascript
chrome.windows.create({url: chrome.extension.getURL('view/popup.html'), type: "detached_panel", height: 480, width: 320});
```

**Comment by mymindstorm** [comment]
I've done 2, 5, and 6. 1 is being tracked in another issue and right now. 4 and 7 are blocked by other underlying sync issues right now, so I'm going to close this for the time being.
